tools/update-pieeprom.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/sh
+
+# Utility to update the EEPROM image (pieeprom.bin) and signature
+# (pieeprom.sig) with a new EEPROM config.
+#
+# pieeprom.original.bin - The source EEPROM from rpi-eeprom repo
+# boot.conf - The bootloader config file to apply.
+
+set -e
+
+script_dir="$(cd "$(dirname "$0")" && pwd)"
+
+# Minimum version for secure-boot support
+BOOTLOADER_SECURE_BOOT_MIN_VERSION=1628600827
+SRC_IMAGE="pieeprom.original.bin"
+CONFIG="boot.conf"
+DST_IMAGE="pieeprom.bin"
+PEM_FILE=""
+TMP_CONFIG_SIG=""
+
+die() {
+   echo "$@" >&2
+   exit ${EXIT_FAILED}
+}
+
+cleanup() {
+   [ -f "${TMP_CONFIG}" ] && rm -f "${TMP_CONFIG}"
+}
+
+usage() {
+cat <<EOF
+    update-pieeprom.sh [key]
+
+    Updates and EEPROM image by replacing the configuration file and generates
+    the new pieeprom.sig file.
+
+    The bootloader configuration may also be signed by specifying the name
+    of an .pem with containing a 2048bit RSA public/private key pair.
+
+    RSA signature support requires the Python Crypto module. To install:
+    python3 -m pip install Crypto
+
+    -c Bootloader config file - default: "${SRC_IMAGE}"
+    -i Source EEPROM image - default: "${CONFIG}"
+    -o Output EEPROM image - default: "${DST_IMAGE}"
+    -k Optional RSA private PEM file - default: "${PEM_FILE}"
+
+The -k argument signs the EEPROM configuration using the specified RSA 2048
```

Comment:
For consistency, `2048-bit` should be before `RSA`, and with a hyphen is the correct grammar.